FiveCalls/FiveCalls/NewUserLocation.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  NewUserLocation.swift
+//  FiveCalls
+//
+//  Created by Nick O'Neill on 8/9/23.
+//  Copyright © 2023 5calls. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import Rswift
+
+// NewUserLocation is like UserLocation from the old app but it doesn't need a singleton to maintain its state
+class NewUserLocation {
+    enum LocationType: String {
+        case address
+        case coordinates
+        
+        init?(rawValue: String) {
+            switch rawValue {
+            //handle legacy persisted zipCodes as addresses
+            case "address", "zipCode":
+                self = .address
+            case "coordinates":
+                self = .coordinates
+            default:
+                return nil
+            }
+        }
+    }
+        
+    var locationType: LocationType
+    var locationValue: String
+    var locationDisplay: String
+    
+    init(address: String, display: String? = nil, gotDisplay: ((NewUserLocation) -> Void)? = nil) {
```

Comment:
no need for changes now, but perhaps we can integrate this with our unidirectional store to observe changes instead of requiring a callback closure? Using Combine to observe changes to `locationDisplay` on the current location object in the store if that's the property we care most about.

Reply:
Definitely, adding some todos